localstack/services/generic_proxy.py

```diff
@@ -147,9 +147,10 @@ def forward(self, method):
             # copy headers and return response
             self.send_response(response.status_code)
             for header_key, header_value in iteritems(response.headers):
-                if header_key.lower() != 'Content-Length'.lower():
+                if header_key.lower() != 'Content-Length'.lower() or method == 'HEAD':
                     self.send_header(header_key, header_value)
-            self.send_header('Content-Length', '%s' % len(response.content))
+            if method != 'HEAD':
```

Comment:
In the (unlikely?) case that the `response` of a `HEAD` request does *not* contain a `Content-Length`, we'll end up sending no `Content-Length` header at all in the final response. Not sure if this is the desired behavior.

To be on the safe side, we could change it to something like this:

```
content_length_sent = False
for header_key, header_value in iteritems(response.headers):
    is_content_length_header = header_key.lower() != 'Content-Length'.lower()
    if is_content_length_header or method == 'HEAD':
        self.send_header(header_key, header_value)
        content_length_sent = content_length_sent or is_content_length_header
if not content_length_sent:
    self.send_header('Content-Length', '%s' % len(response.content))
```